tests/test_stac.py

```diff
@@ -9,3 +12,22 @@ def test_create_collection(self):
 
         collection.set_self_href("http://example.com/collection.json")
         collection.validate()
+
+    def test_create_item(self):
+        item = create_item(
+            "al",
+            "2011",
+            test_data.get_path("data-files/m_3008501_ne_16_1_20110815-downsampled.tif"),
+            test_data.get_path("data-files/m_3008501_ne_16_1_20110815.txt"),
+        )
+
+        image_asset = item.assets["image"].to_dict()
+        raster_bands = image_asset["raster:bands"]
+        self.assertEqual(len(raster_bands), 4)
+        for raster_band in raster_bands:
+            self.assertEqual(raster_band["nodata"], 0)
+            self.assertEqual(raster_band["spatial_resolution"], item.properties["gsd"])
+            self.assertEqual(raster_band["data_type"], DataType.UINT8)
+            self.assertEqual(raster_band["unit"], "none")
+
+        self.assertEqual(item.properties["sci:doi"], "10.5066/F7QN651G")
```

Comment:
It'd be better if we used the pystac extensions to access these values -- that provides some extra validation on the fields, checks for the `stac_extensions` link, etc. So something like:

```python
raster = RasterExtension.ext(image_asset)
// assert stuff
scientific = ItemScientificExtension.ext(item)
// assert stuff
```